homeassistant/components/alarm_control_panel/group.py

```diff
@@ -0,0 +1,241 @@
+        def group_alarm_startup(event):
+            """Update template on startup."""
+            async_track_state_change(
+                self.hass, self._entity_ids, group_alarm_state_listener)
+
+            self.async_schedule_update_ha_state(True)
+
+        self.hass.bus.async_listen_once(
+            EVENT_HOMEASSISTANT_START, group_alarm_startup)
+
+    @asyncio.coroutine
+    def _async_send_message(self, service, **kwargs):
+        """Send message to all entities in the group."""
+        payload = {key: val for key, val in kwargs.items() if val}
+
+        tasks = []
+        for entity_id in self._entity_ids:
+            sending_payload = deepcopy(payload.copy())
+            sending_payload[ATTR_ENTITY_ID] = entity_id
+            tasks.append(self.hass.services.async_call(
+                DOMAIN, service, sending_payload))
+
+        if tasks:
+            yield from asyncio.wait(tasks, loop=self.hass.loop)
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def should_poll(self):
+        """No polling needed."""
+        return False
+
+    @property
+    def code_format(self):
+        """The code format is customizable."""
+        return self._code_format
+
+    @asyncio.coroutine
+    def async_alarm_disarm(self, code=None):
+        """Send disarm command."""
+        yield from self._async_send_message(SERVICE_ALARM_DISARM,
+                                            code=code)
+
+    @asyncio.coroutine
+    def async_alarm_arm_home(self, code=None):
+        """Send arm home command."""
+        yield from self._async_send_message(SERVICE_ALARM_ARM_HOME,
+                                            code=code)
+
+    @asyncio.coroutine
+    def async_alarm_arm_away(self, code=None):
+        """Send arm away command."""
+        yield from self._async_send_message(SERVICE_ALARM_ARM_AWAY,
+                                            code=code)
+
+    @asyncio.coroutine
+    def async_alarm_arm_night(self, code=None):
+        """Send arm night command."""
+        yield from self._async_send_message(SERVICE_ALARM_ARM_NIGHT,
+                                            code=code)
+
+    @asyncio.coroutine
+    def async_alarm_trigger(self, code=None):
+        """Send alarm trigger command."""
+        yield from self._async_send_message(SERVICE_ALARM_TRIGGER,
+                                            code=code)
+
+    @asyncio.coroutine
+    def async_alarm_arm_custom_bypass(self, code=None):
+        """Send arm custom bypass command."""
+        yield from self._async_send_message(SERVICE_ALARM_ARM_CUSTOM_BYPASS,
+                                            code=code)
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        # _update_state always behaves as if a transition was in progress.
+        # But if the pre- and post-state match, we're not "pending"!
+        self._update_state()
```

Comment:
I would move all this logic to `_update_state` which should be renamed to `async_update`. Just return `self._state` here.